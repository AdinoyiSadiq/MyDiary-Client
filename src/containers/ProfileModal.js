import React, { Component } from 'react';
import { connect } from 'react-redux';
import ProfileToggle from '../components/ProfileToggle';
import ProfileDetails from '../components/ProfileDetails';
import { getUserProfile } from '../actions/profileActions';

class ProfileModal extends Component {
  componentDidMount = () => {
    const { getUserProfile } = this.props;
    getUserProfile();
  }

  renderProfileModal() {
    const { profile } = this.props;
    if (profile) {
      return (
        <section>
          <div>
            <ProfileToggle />
            <ProfileDetails profile={profile} />
          </div>
        </section>
      );
    } else {
      return <div>Loading...</div>
    }
  }

  render() {
    return (
      <main>
        {this.renderProfileModal()}
	    </main>
    );
  }
}

function mapStateToProps(state) {
  return { profile: state.userProfile.profile };
}

export default connect(mapStateToProps, { getUserProfile })(ProfileModal);