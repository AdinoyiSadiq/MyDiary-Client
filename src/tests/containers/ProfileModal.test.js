import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import ProfileModal from '../../containers/ProfileModal';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <ProfileModal />
      </MemoryRouter>
    </Root>,
  );
});

describe('Profile Modal UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});  