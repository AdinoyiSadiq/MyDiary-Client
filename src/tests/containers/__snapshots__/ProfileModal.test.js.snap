// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Modal UI render features container should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(ProfileModal)>
          <ProfileModal
            getUserProfile={[Function]}
          >
            <main>
              <div>
                Loading...
              </div>
            </main>
          </ProfileModal>
        </Connect(ProfileModal)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
